editor/src/clj/editor/cljfx_form_view.clj

```diff
@@ -1298,74 +1325,152 @@
 
 ;; region jump-to
 
-(defmethod handle-event :jump-to [{:keys [section ^Node parent]}]
-  (let [^ScrollPane scroll-pane (.lookup parent "#scroll-pane")
-        ^Node node (.lookup parent (str "#" section))
-        content-height (-> scroll-pane .getContent .getBoundsInLocal .getHeight)
-        viewport-height (-> scroll-pane .getViewportBounds .getHeight)]
-    (when (> content-height viewport-height)
-      (.setVvalue scroll-pane (/ (- (.getMinY (.getBoundsInParent node))
-                                    24)                     ;; form padding
-                                 (- content-height viewport-height))))))
-
-(defn- jump-to-button [{:keys [visible-titles]}]
-  {:fx/type fx.menu-button/lifecycle
-   :text "Jump to..."
-   :style-class "jump-to-menu-button"
-   :disable (empty? visible-titles)
-   :items (->> visible-titles
-               (sort #(.compareToIgnoreCase ^String %1 %2))
-               (mapv (fn [title]
-                       {:fx/type fx.menu-item/lifecycle
-                        :on-action {:event-type :jump-to
-                                    :section (section-id title)}
-                        :text title})))})
+(defmethod handle-event :jump-to [{:keys [section ui-state]}]
+  {:set-ui-state (assoc ui-state :selected-section-title section)})
+
+(def ext-ensure-visible
+  (fx/make-ext-with-props
+    {:ensure-visible
+     (fx.prop/make
+       (fx.mutator/setter
+         (fn [^ScrollPane pane ^Node child]
+           (when child
+             (let [content-height (-> pane .getContent .getBoundsInLocal .getHeight)
+                   viewport-bounds (.getViewportBounds pane)
+                   viewport-height (.getHeight viewport-bounds)
+                   viewport-bottom (- (.getHeight viewport-bounds) (.getMinY viewport-bounds))
+                   child-bounds (-> pane .getContent (.sceneToLocal (.localToScene child (.getBoundsInLocal child))))]
+               (when (< viewport-height content-height)
+                 (cond
+                   ;; when child view is below viewport, scroll down
+                   (< viewport-bottom (.getMaxY child-bounds))
+                   (.setVvalue pane (/ (- (.getMaxY child-bounds) (.getHeight viewport-bounds))
+                                       (- content-height viewport-height)))
+                   ;; when child view is above viewport, scroll down
+                   (< (.getMinY child-bounds) (- (.getMinY viewport-bounds)))
+                   (.setVvalue pane (/ (.getMinY child-bounds)
+                                       (- content-height viewport-height)))))))))
+       fx.lifecycle/dynamic)}))
```

Comment:
This seems like it could be of general use? Would it make sense to move it to `editor.fxui`?

Reply:
Yes!